File: Latihan/Latihan2/Without Library/Java/ManualQueue/src/Queue.java
public class Queue {
    private int maxSize;
    private Object[] queueArray;
    private int front;
    private int rear;
    private int currentSize;

    public Queue(int s) {
        maxSize = s;
        queueArray = new Object[maxSize];
        front = 0;
        rear = -1;
        currentSize = 0;
    }

    public void enqueue(Object item) {
        if (isFull()) {
            System.out.println("Queue is full");
            return;
        }
        rear++;
        queueArray[rear] = item;
        currentSize++;
    }

    public Object dequeue() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return null;
        }
        Object item = queueArray[front];
        front++;
        currentSize--;
        return item;
    }

    public Object element() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return null;
        }
        return queueArray[front];
    }

    public Object peek() {
        if (isEmpty()) {
            return null;
        }
        return queueArray[front];
    }

    public Object poll() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return null;
        }
        Object item = queueArray[front];
        front++;
        currentSize--;
        return item;
    }

    public boolean isFull() {
        return currentSize == maxSize;
    }

    public boolean isEmpty() {
        return currentSize == 0;
    }

    public int size() {
        return currentSize;
    }
}

File: Latihan/Latihan2/Without Library/Python/Queue.py
class Queue:
    def __init__(self, s):
        self.maxSize = s
        self.queueArray = [None] * self.maxSize
        self.front = 0
        self.rear = -1
        self.currentSize = 0

    def enqueue(self, item):
        if self.isFull():
            print("Queue is full")
            return
        self.rear += 1
        self.queueArray[self.rear] = item
        self.currentSize += 1

    def dequeue(self):
        if self.isEmpty():
            print("Queue is empty")
            return None
        item = self.queueArray[self.front]
        self.front += 1
        self.currentSize -= 1
        return item

    def element(self):
        if self.isEmpty():
            print("Queue is empty")
            return None
        return self.queueArray[self.front]

    def peek(self):
        if self.isEmpty():
            return None
        return self.queueArray[self.front]

    def poll(self):
        if self.isEmpty():
            print("Queue is empty")
            return None
        item = self.queueArray[self.front]
        self.front += 1
        self.currentSize -= 1
        return item

    def isFull(self):
        return self.currentSize == self.maxSize

    def isEmpty(self):
        return self.currentSize == 0

    def size(self):
        return self.currentSize

queue = Queue(5)
queue.enqueue("Java")
queue.enqueue("DotNet")
queue.enqueue("PHP")
queue.enqueue("HTML")

print(queue.dequeue())
print(queue.element())
print(queue.poll())
print(queue.peek())